### PR TITLE
[DR-1251] Add validations fluid into Billing Page

### PR DESCRIPTION
**Pull request by mvillaseco**
# Overview
The needed changes to apply new validations into billing page.

# Changes
 - Fix test.
 - Remove old manually validations.
 - Change locales keys.
 - Fix issue in Responsive.

### CODE REVIEW

File: src/wwwroot/controllers/BillingCtrl.js [hunk 1]
```diff
@@ -118,12 +118,21 @@
...
         var month = input.$modelValue.slice(0,2);
       }
+      if (input.$modelValue && input.$modelValue.length == 4) {
+        var currentDate = new Date();
+        var currentYear = currentDate.getFullYear().toString().slice(2);
+        var currentMonth = currentDate.getMonth() + 1;
+        if (input.$modelValue) {
+          var year = input.$modelValue.slice(2);
+          var month = input.$modelValue.slice(0,2);
+        }
```
Comment (andresmoschini) [inline src/wwwroot/controllers/BillingCtrl.js]: Why did you added previous lines?

File: src/wwwroot/partials/settings/billing.html [hunk 2]
```diff
@@ -26,45 +26,40 @@ <h3 class="special">{{ 'billing_address_title' | translate }}</h3>
               <label class="special uppercase" for="name">{{'downloads_grid_name_header' | translate}}</label>
             </div>
             <div class="flex three flex--wrap">
-              <input id="name" name="name" ng-model="vm.name" ng-class="(vm.submitted || form.name.$dirty) && form.name.$invalid ? 'error' : ''" class="input input--large" required />
-              <div class="width--full animate-error-container" ng-class="(vm.submitted || form.name.$dirty) && form.name.$invalid ? 'show' : 'hide'">
-                <span class="input--error" ng-show="(vm.submitted || form.name.$dirty) && form.name.$error.required"><i class="arrow--up"></i>{{'validation_error_required' | translate}}</span>
-              </div>
+              <input id="name" name="name" ng-model="vm.name" class="input input--large" validation-errors-fluid required/>
```
Comment (andresmoschini) [inline src/wwwroot/partials/settings/billing.html]: 😍 Wow! look at this! I love it

File: src/wwwroot/partials/settings/billing.html [hunk 3]
```diff
@@ -26,45 +26,40 @@ <h3 class="special">{{ 'billing_address_title' | translate }}</h3>
...
           <div class="item flex">
             <div class="flex one flex--vertically-start">
               <label class="special uppercase" for="company">{{'company_text' | translate}}</label>
             </div>
             <div class="flex three flex--wrap">
-              <input id="company" name="company" ng-model="vm.company" ng-class="(vm.submitted || form.company.$dirty) && form.company.$invalid ? 'error' : ''" class="input input--large" required />
-              <div class="width--full animate-error-container" ng-class="(vm.submitted || form.company.$dirty) && form.company.$invalid ? 'show' : 'hide'">
-                <span class="input--error" ng-show="(vm.submitted || form.company.$dirty) && form.company.$error.required"><i class="arrow--up"></i>{{'validation_error_required' | translate}}</span>
-              </div>
+              <input id="company" name="company" ng-model="vm.company" class="input input--large" validation-errors-fluid required />
```
Comment (andresmoschini) [inline src/wwwroot/partials/settings/billing.html]: 💘

File: src/wwwroot/partials/settings/billing.html [hunk 4]
```diff
@@ -26,45 +26,40 @@ <h3 class="special">{{ 'billing_address_title' | translate }}</h3>
...
           <div class="item flex">
             <div class="flex one flex--vertically-start">
               <label class="special uppercase" for="address">{{'address_text' | translate}}</label>
             </div>
             <div class="flex three flex--wrap">
-              <input id="address" name="address" ng-model="vm.address" ng-class="(vm.submitted || form.address.$dirty) && form.address.$invalid ? 'error' : ''" class="input input--large" required />
-              <div class="width--full animate-error-container" ng-class="(vm.submitted || form.address.$dirty) && form.address.$invalid ? 'show' : 'hide'">
-                <span class="input--error" ng-show="(vm.submitted || form.address.$dirty) && form.address.$error.required"><i class="arrow--up"></i>{{'validation_error_required' | translate}}</span>
-              </div>
+              <input id="address" name="address" ng-model="vm.address" class="input input--large" validation-errors-fluid required />
```
Comment (andresmoschini) [inline src/wwwroot/partials/settings/billing.html]: 💘  (I cannot stop)

File: src/wwwroot/partials/settings/billing.html [hunk 5]
```diff
@@ -26,45 +26,40 @@ <h3 class="special">{{ 'billing_address_title' | translate }}</h3>
...
+                <div class="flex one flex--vertically-start flex--wrap">
+                  <input id="city" name="city" ng-model="vm.city"class="input input--large" validation-errors-fluid required />
+                </div>
+                <div class="flex one flex--vertically-start zip--container">
+                  <div class="flex one flex--vertically-start flex--wrap label">
+                    <label class="special uppercase" for="zCode">{{'zip_code_label' | translate}}</label>
+                  </div>
+                  <div class="flex three flex--vertically-start flex--wrap">
+                    <input id="zCode" name="zCode" ng-model="vm.zCode" validation-errors-fluid class="input input--small width--half" required />
+                  </div>
```
Comment (andresmoschini) [inline src/wwwroot/partials/settings/billing.html]: You have also changed the structure, only for curiosity, why?

File: src/wwwroot/partials/settings/billing.html [hunk 6]
```diff
@@ -103,54 +95,44 @@ <h3 class="special">{{ 'payment_method_title' | translate }}</h3>
...
-            </svg>
-            <svg class="icon icon-master">
-              <use xlink:href="/images/sprite.svg#doppler-icon-master-card"></use>
-            </svg>
-            <svg class="icon icon-amex">
-              <use xlink:href="/images/sprite.svg#doppler-icon-american-card"></use>
-            </svg>
-              <div class="width--full animate-error-container" ng-class="(vm.submitted || form.cardNumber.$dirty) && form.cardNumber.$invalid ? 'show' : 'hide'">
-                <span class="input--error" ng-show="(vm.submitted || form.cardNumber.$dirty) && form.cardNumber.$error.required || form.cardNumber.$error.mask"><i class="arrow--up"></i>{{'validation_error_required' | translate}}</span>
-                <span class="input--error" ng-show="(vm.submitted || form.cardNumber.$dirty) && form.cardNumber.$error.ilegal_number && !form.cardNumber.$error.required && ! form.cardNumber.$error.mask"><i class="arrow--up"></i>{{'validation_error_invalid_card_number' | translate}}</span>
```
Comment (andresmoschini) [inline src/wwwroot/partials/settings/billing.html]: 😍 😍 😍 😍  This is the best one! you have removed two ugly elements, not just one!

**Comment by andresmoschini** [comment]
Thank you very much @mvillaseco, it will help a lot when we create new pages.

File: src/wwwroot/controllers/BillingCtrl.js (hunk 1 above)
Comment (mvillaseco) [inline src/wwwroot/controllers/BillingCtrl.js]: Don't know why hahah, i will merge the two "if" using the one from line 125

File: src/wwwroot/partials/settings/billing.html (hunk 5 above)
Comment (mvillaseco) [inline src/wwwroot/partials/settings/billing.html]: The old validation added manually doesn't had a container, and now it's going to have one, that's why i have to change the sctructure of the html.

**Comment by nreal** [comment]
687474703a2f2f692e696d6775722e636f6d2f436b6f436f44752e676966

**Pull request merged**